### ISSUE TITLE
New user documention "bumps"

### ISSUE THREAD

**Issue by wmakley** [sveltejs/kit#9589]
### Describe the problem

I am learning svelte and sveltekit from scratch, and the introductory text under "Getting Started" and "Core Concepts" leaves me with several mysteries that are requiring me to leave the docs and start googling for understanding to build a "real" app. Thanks for listening.

1. The biggest: Where do I put my components and how do I import them in a "route"? What does a page or large component with multiple nested components look like? "Project Structure" is where I would expect to find this information.
2. What the heck is "+" anyway? This is very conspicuous in the routing docs and I was hoping it would be explained there.
3. The adapters page does not make it immediately clear that I should visit the page **for the specific adapter** to learn how to install and switch to it, and that the contents of https://kit.svelte.dev/docs/adapters are mostly a stub. I was stuck here for a little while.

Thank you!

### Describe the proposed solution

Some additional text and examples in the above pages to get new users to the point they can actually build something without hitting brick walls.

### Alternatives considered

Googling instead of using the official docs.

### Importance

would make my life easier

**Comment by gtm-nayan** [comment]
For 1, the project structure docs contain
> `lib` contains your library code (utilities and components)
> `routes` contains the [routes](https://kit.svelte.dev/docs/routing) of your application. You can also colocate other components that are only used within a single route here

For 2, the Routing docs say
> Each route directory contains one or more route files, which can be identified by their + prefix.

@wmakley Any specific improvements you can think of that'd make it clearer?

**Comment by tcc-sejohnson** [comment]
@wmakley In addition to @gtm-nayan's questions, you may find it friendlier to start with learn.svelte.dev -- the docs _are_ good for learning, but they're first and foremost a reference. The tutorial is meant for teaching. 😁

**Comment by fen89** [comment]
@wmakley 

**Where  do I put my components?**

**tl;dr**: Wherever you want them to be. 


It depends (on your preferences, your needs, etc), but let me show you some possibilities:

a) Flat components folder:
```
/lib
	/components
		button.svelte
		input-text.svelte
		input-number.svelte
		form.svelte
		user-list.svelte
		user-filter.svelte
		dashboard-widget.svelte
		[...]
```

b) Feature driven (with shared/common folder) components folder:
```
/lib
	/components
		/common
			button.svelte
			input-text.svelte
			input-number.svelte
			form.svelte
		/user
			user-list.svelte
			user-filter.svelte
		/dashboard
			dashboard-widget.svelte
		[...]
```

c) Co-located route components:

Stick with subcomponents co-related to your routes, Structure is up to you, I am happy with the following.

```
/lib
	/components
		button.svelte
		input-text.svelte
		input-number.svelte
		form.svelte
		[...]
/routes
	/users
		/ui
			user-list.svelte
			user-filter.svelte
		/utils
			concat-array.ts
		/data-access
			user.store.ts
			user-filter.store.ts
		+page.svelte
		[...]
	/dashboard
		/ui
			dashboard-widget.svelte
		/utils
		/data-access
		+page.svelte
		[...]
```

Keep in mind that the shared components (like for forms, buttons, etc) would still live inside $lib/components:

The modular folder approach with co-located subcomponents is the easiest to grasp for me and others in various teams I worked with.

**How do I import them into my routes?**

You need to directly import them from their corresponding location inside your project. E.g. if the components are located inside `lib/components` you can use the prefix `$lib` which resolves to the `libs` directory` to access these files. The following snippet would import a `Button` component which locates inside `/libs/components`:

+page.svelte
``` 
<script lang="ts">
 import Button from '$lib/components/button.svelte';
</script>

<Button>Click me</Button>
 ```

button.svelte
``` 
<button>
   <slot />
</button>
```

If you want to import a component which lives near your routes (as explained above) you import it with a relative path:

+page.svelte
``` 
<script lang="ts">
 import UserList from './ui/user-list.svelte';
</script>

<UserList />
 ```
